apps/ledger-live-mobile/src/screens/SyncOnboarding/index.tsx

```diff
@@ -207,6 +207,15 @@ export const SyncOnboarding = ({
     stopPolling,
   });
 
+  // Unmount cleanup to make sure the polling is stopped.
+  // The cleanup function triggered by the useEffect of useOnboardingStatePolling
+  // has been observed to be called after, and some apdu could still be exchanged with the device
+  useEffect(() => {
+    return () => {
+      setStopPolling(true);
+    };
+  }, []);
+
```

Comment:
isn't this something that can be moved inside the useOnboardingStatePolling hook? 
I mean the state variable and the cleanup useEffect. It doesn't seem that "stopPolling" is used anywhere else except to pass to the hook itself.

Reply:
For the state variable: the `stopPolling` is used in the component to stop the polling after a certain step, the polling is not activated all the time during the onboarding happening on the screen, so i would say it is necessary

For the cleanup useEffect: there is already a clean up on `useOnboardingStatePolling` hook, but, from my observation with logs, when returning back to the BLE scanning/pairing screen for ex, the useEffect cleanup function is called a bit after the BLE screen gets focus (why ? idk, but i don't think it's coming from our code) which triggers the BLE scanning while the sync onboarding was not yet cleaned

Forcing the hook to clean up by setting `stopPolling` works because the state of the `SyncOnboarding` component is then updated before react-navigation moves to the next screen.

That what i tried to explain in the added comment:
>  // Unmount cleanup to make sure the polling is stopped.
  // The cleanup function triggered by the useEffect of useOnboardingStatePolling
  // has been observed to be called after, and some apdu could still be exchanged with the device

Reply:
Got it!